clusterloader2/pkg/measurement/common/profile.go

```diff
@@ -61,9 +61,14 @@ func (p *profileMeasurement) populateProfileConfig(config *measurement.Measureme
 	if p.config.provider, err = util.GetStringOrDefault(config.Params, "provider", config.ClusterFramework.GetClusterConfig().Provider); err != nil {
 		return err
 	}
-	if p.config.host, err = util.GetStringOrDefault(config.Params, "host", config.ClusterFramework.GetClusterConfig().GetMasterIp()); err != nil {
+	masterIPs, err := util.GetStringSliceOrDefault(config.Params, "hosts", config.ClusterFramework.GetClusterConfig().GetMasterIps())
+	if err != nil {
 		return err
 	}
+	if masterIPs == nil || len(masterIPs) < 1 {
+		return fmt.Errorf("empty slice: there are no master IPs registered %v", masterIPs)
+	}
+	p.config.host = masterIPs[0]
```

Comment:
Same here and in other files.

Also, it might be a good idea to tackle each of these files in a separate PR (depending on how much work needs to be done to rewrite a single one). If it's more than 50 lines per file then it will better to split them into separate PRs (it will be easier / faster to review).